colabs/Zmiana_typu_danych.ipynb

```diff
@@ -0,0 +1,1145 @@
+        "!gzip -dc /content/title.basics.tsv.gz > title.basics.csv\n",
+        "!gzip -dc /content/title.crew.tsv.gz > title.crew.csv\n",
+        "!gzip -dc /content/title.episode.tsv.gz > title.episode.csv\n",
+        "!gzip -dc /content/title.principals.tsv.gz > title.principals.csv\n",
+        "!gzip -dc /content/title.ratings.tsv.gz > title.ratings.csv"
+      ],
+      "metadata": {
+        "id": "IpXAjtV4XeKx"
+      },
+      "execution_count": 92,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "#Wczytanie danych"
+      ],
+      "metadata": {
+        "id": "pPWxy5j_aEvo"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "##Ustalenie typów danych"
+      ],
+      "metadata": {
+        "id": "JBOGZE2nWv44"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from pyspark.sql.types import StructType, StringType, IntegerType, BooleanType, FloatType, TimestampType, DateType, ArrayType, MapType\n",
+        "from typing import List, Tuple, Dict, Any\n",
+        "from pyspark.sql.types import StructType, StringType"
+      ],
+      "metadata": {
+        "id": "KW0lFd6HfDqU"
+      },
+      "execution_count": 93,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "map_types = {\n",
+        "    str : StringType(),\n",
+        "    int : IntegerType(),\n",
+        "    bool : BooleanType(),\n",
+        "    float: FloatType(),\n",
+        "    'timestamp' : TimestampType(),\n",
+        "    'date' : DateType(),\n",
+        "    List[str] : ArrayType(StringType()),\n",
+        "    Tuple[str] : ArrayType(StringType()),\n",
+        "    Dict[str, str] : MapType(StringType(), StringType())\n",
+        "}"
+      ],
+      "metadata": {
+        "id": "CqMSVkG6fFYK"
+      },
+      "execution_count": 94,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "column_type_collection = {\n",
+        "    int : ['ordering', 'startYear', 'endYear', 'runtimeMinutes', 'seasonNumber', 'episodeNumber', 'numVotes', 'birthYear', 'deathYear' ],\n",
+        "    str : ['titleId', 'title', 'region', 'language', 'types', 'attributes', 'tconst', 'titleType', 'primaryTitle', 'originalTitle', 'genres', 'directors', 'writers', 'parentTconst', 'category', 'job', 'characters', 'primaryName', 'primaryProfession', 'knownForTitles'],\n",
+        "    bool : ['isOriginalTitle', 'isAdult' ],\n",
+        "    float : ['averageRating']\n",
+        "}\n",
+        "#ustalenie danych typu array(string) w ten sposób nie jest możliwe, więc ten typ zostanie zmieniony później"
+      ],
+      "metadata": {
+        "id": "sOhcCmU2fNWU"
+      },
+      "execution_count": 95,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "pola_title_akas = '''titleId,ordering,title,region,language,types,attributes,isOriginalTitle '''\n",
```

Comment:
To wygląda jak lenistwo by tego nie podzielić
```suggestion
        "pola_title_akas = ['titleId','ordering','title','region','language','types','attributes','isOriginalTitle']\n",
```

Reply:
zmienione